testing/opae-c/dummy_plugin.c

```diff
@@ -0,0 +1,203 @@
+// Copyright(c) 2018-2019, Intel Corporation
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+#include <stdio.h>
+#include <json-c/json.h>
+#include "adapter.h"
+#include "opae_int.h"
+
+#define DUMMY_HIDDEN __attribute__((visibility("hidden")))
+
+static uint32_t _fake_tokens;
+
+int DUMMY_HIDDEN dummy_plugin_initialize(void)
+{
+	return 0;
+}
+
+int DUMMY_HIDDEN dummy_plugin_finalize(void)
+{
+	return 0;
+}
+
+bool DUMMY_HIDDEN dummy_plugin_supports_device(const char *device_type)
+{
+	UNUSED_PARAM(device_type);
+	return true;
+}
+
+bool DUMMY_HIDDEN dummy_plugin_supports_host(const char *hostname)
+{
+	UNUSED_PARAM(hostname);
+	return true;
+}
+
+typedef struct _dummy_token {
+	int number;
+} dummy_token;
+
+typedef struct _dummy_handle {
+	int number;
+	dummy_token *token;
+} dummy_handle;
+
+#define MIN(x,y) (x < y) ? x : y
+fpga_result DUMMY_HIDDEN dummy_plugin_fpgaEnumerate(const fpga_properties *filters,
+						    uint32_t num_filters, fpga_token *tokens,
+						    uint32_t max_tokens, uint32_t *num_matches)
+{
+	uint32_t i = 0;
+	uint32_t min = MIN(_fake_tokens, max_tokens);
+	UNUSED_PARAM(filters);
+	UNUSED_PARAM(num_filters);
+	*num_matches = _fake_tokens;
+
+	for ( ; i < min; ++i) {
+		dummy_token *t = (dummy_token*)malloc(sizeof(dummy_token));
+		t->number = i;
+		tokens[i] = t;
```

Comment:
This always assumes that tokens is non-NULL.  I assume this is OK for the tests.

Reply:
fixed